tests/dask/test_IP_jvecjtvecadj_dask.py

```diff
@@ -1,23 +1,137 @@
+import os
+import shutil
+import tarfile
 import unittest
-import discretize
+
+import discretize as ds
 import numpy as np
 
 import SimPEG.dask  # noqa: F401
-from SimPEG import maps
-from SimPEG import data_misfit
-from SimPEG import regularization
-from SimPEG import optimization
-from SimPEG import inversion
-from SimPEG import inverse_problem
-from SimPEG import tests
-
-from SimPEG.electromagnetics import resistivity as dc
+from SimPEG import (
+    data_misfit,
+    inverse_problem,
+    inversion,
+    maps,
+    optimization,
+    regularization,
+    tests,
+    utils,
+)
 from SimPEG.electromagnetics import induced_polarization as ip
-import shutil
+from SimPEG.electromagnetics import resistivity as dc
+from SimPEG.utils.io_utils.io_utils_electromagnetics import read_dcip2d_ubc
 
 np.random.seed(30)
 
 
+class IPProblemTests2DN(unittest.TestCase):
```

Comment:
I know having more tests is good (but the testing sweet is already quite large and time consuming), What specifically does this test that is not already tested?

Reply:
I should put it forward higher (comment on line 88 right now)

- The goal is to make a test without calling` make_synthetic_data` first to simulate a real data case. 

- `make_synthetic_data` was the reason why the issue being fixed here was not detected by the tests.

- This test is built by taking the already existing 2D files used in the [IP2D tutorial](https://docs.simpeg.xyz/content/tutorials/06-ip/plot_inv_2_dcip2d.html), with a much smaller mesh (1600 cells).

- I also did not see any test for IP2D + Dask (all IP tests with Dask are for the 3D case). This added test covers this combination.

Reply:
added some docstring / comments

Reply:
I think more so the underlying issue was that the simulation failed when not explicitly calling `fields` before any `Jvec`, `JTvec` and in the dask case `getJTJdiag`.

Reply:
Indeed, that is the core of the issue